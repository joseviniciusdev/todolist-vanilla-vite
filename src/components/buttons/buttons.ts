export const disableButtonClick = (el: HTMLButtonElement) => {
    el.addEventListener('click', function() {
        el.setAttribute('disabled', '')
    })
}