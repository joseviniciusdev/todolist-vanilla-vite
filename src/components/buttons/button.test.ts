import { test, expect } from "vitest";
import { disableButtonClick } from "../buttons/buttons";
import { JSDOM } from "jsdom";
const dom = new JSDOM();

test("add attr Disabled", () => {

	const dom = new JSDOM()
	global.document = dom.window.document
	// global.window = dom.window

	document.body.innerHTML = '<button id="button">event</button>';

	const button = document.querySelector<HTMLButtonElement>("#button");

	if (button) {
		disableButtonClick(button);
		button.click();
	}

	expect(document.body).toMatchSnapshot();
});
