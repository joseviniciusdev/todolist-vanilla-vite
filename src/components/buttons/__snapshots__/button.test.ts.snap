// Vitest Snapshot v1

exports[`add attr Disabled 1`] = `
<body>
  <button
    disabled=""
    id="button"
  >
    event
  </button>
</body>
`;
